### PR TITLE
Update Column API docs

### PR DESCRIPTION
**Pull request by dbeatty10**
## What are you changing in this pull request and why?

Updates to docs inspired by this feature request: https://github.com/dbt-labs/dbt-core/issues/7103

Updates made:
1. The docs gave `(dtype, size)` as the parameters for `numeric_type()`, but it should be [`(dtype, precision, scale)`](https://github.com/dbt-labs/dbt-core/blob/72076b3fe5a9a97884bbc4b6951a984eb50556a9/core/dbt/adapters/base/column.py#L111-L118) instead.
1. Added a section for using columns to match the section for using relations
1. Added `col.numeric_type` to match `col.string_type` in the column Python example usage

## Previews

Here are the previews of each of the sections that were updated:
1. https://deploy-preview-2949--docs-getdbt-com.netlify.app/reference/dbt-classes#static-methods
1. https://deploy-preview-2949--docs-getdbt-com.netlify.app/reference/dbt-classes#using-columns
1. https://deploy-preview-2949--docs-getdbt-com.netlify.app/reference/dbt-classes#column

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.

### CODE REVIEW

File: website/docs/reference/dbt-classes.md [hunk 1]
```diff
@@ -114,7 +115,53 @@ col.string_type() # character varying (255)
 
 ### Static methods
 - **string_type(size)**:  Returns a database-useable representation of the string type (eg. `character varying(255)`)
-- **numeric_type(dtype, size)**: Returns a database-useable representation of the numeric type (eg. `numeric(12, 4)`)
+- **numeric_type(dtype, precision, scale)**: Returns a database-useable representation of the numeric type (eg. `numeric(12, 4)`)
+
+### Using Columns
```
Comment (mirnawong1) [inline website/docs/reference/dbt-classes.md]: sentence case per style guide

```suggestion
### Using columns
```

**Review by mirnawong1 — DISMISSED** [review]
just one small sentence casing recommendation @dbeatty10 ! approved and happy to reapproved as well. 

one non blocker question - is it correct that the file name isn't snake_case? for example, `Column Usage.sql.`?

**Comment by dbeatty10** [comment]
> just one small sentence casing recommendation @dbeatty10 ! approved and happy to reapproved as well.
> 
> one non blocker question - is it correct that the file name isn't snake_case? for example, `Column Usage.sql.`?

Thanks for this feedback @mirnawong1 !

In https://github.com/dbt-labs/docs.getdbt.com/pull/2949/commits/2565b634fb3e144b797854fe4780a6f5a09e005f made two types of updates:
1. More sentence casing updates in headers and code comments
1. Use snake_case for two different file names

**Review by mirnawong1 — APPROVED** [review]
@dbeatty10 thank you so much, always love your thorough pr's and explanation 🙏 - all ready for you to merge ✨ !

**Pull request merged**